src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs

```diff
@@ -134,10 +131,9 @@ protected virtual void Disconnected(JsonRpcDisconnectedEventArgs e)
             }
             catch (Exception ex) // no when since Extensions.InvokeAsync already recorded it
             {
-                // any exception can be thrown from StreamJsonRpc if JsonRpc is disposed in the middle of read/write.
-                // until we move to newly added cancellation support in JsonRpc, we will catch exception and translate to
-                // cancellation exception here. if any exception is thrown unrelated to cancellation, then we will rethrow
-                // the exception
+                // StreamJsonRpc throws RemoteInvocationException if the call is cancelled.
```

Comment:
Can we extract this pattern (this throw, the log error, then the throw of our own cancellation) into a single function?